addons/mass_mailing/models/mailing.py

```diff
@@ -493,6 +491,15 @@ def write(self, values):
 
         return result
 
+    def _add_missing_campaigns_for_ab_testing(self):
+        """  Compute the values of the A/B test campaigns for the mailings that do not have
+        campaigns set already """
+        campaign_vals = [
+            mailing._get_default_ab_testing_campaign_values()
+            for mailing in self.filtered(lambda mailing: not mailing.campaign_id)
```

Comment:
You could add an extra check to "ab_testing_enabled" in the filtered lambda and then you would be able to call this method in the "create" as well (I think), which would be cleaner.